### PR TITLE
added topics query to api/articles

### PR DESCRIPTION
**Pull request by lw1066**

### CODE REVIEW

File: __tests__/endpoint.tests.js [hunk 1]
```diff
@@ -103,6 +103,22 @@ describe('GET/api/articles', () => {
...
     })
+    it('200: accepts a topic query which filters articles by topic', async () => {
+        const result = await request(app).get('/api/articles?topic=cats');
+        const expectedColumns = ['article_id', 'title', 'topic', 'author', 'created_at', 'votes', 'article_img_url', 'comment_count'];
+
+        expect(result.status).toBe(200);
+        expect(result.body.articles).toHaveLength(1);
+        expect(result.body.articles).toBeSortedBy('created_at', {descending:true});
+        expect(Object.keys(result.body.articles[0])).toEqual(expectedColumns);
+    })
```
Comment (Harrison365) [inline __tests__/endpoint.tests.js:114]: We need to check that the articles coming back are about cats.

Comment (Harrison365) [reply __tests__/endpoint.tests.js:114]: We will also need a test for a topic that exists but returns an empty array as it has no articles.

File: __tests__/endpoint.tests.js [hunk 2]
```diff
@@ -103,6 +103,22 @@ describe('GET/api/articles', () => {
...
+    it('404: returns not found if given a topic which no articles use', async () => {
+        const result = await request(app).get('/api/articles?topic=notatopic');
+
+        expect(result.status).toBe(404);
+        const err = result.body;
+        expect(err.msg).toBe('No articles for topic: notatopic');
+    })
```
Comment (Harrison365) [inline __tests__/endpoint.tests.js:121]: Not a topic 👍 
No articles or this topic shouldn't throw an error if the topic exits.

Thanks :)

File: models/select-all-articles.model.js [hunk 3]
```diff
@@ -12,16 +13,26 @@ exports.selectAllArticles = async () => {
...
+    }
+
+    queryString += ` GROUP BY 
             articles.article_id, 
             articles.title, 
             articles.topic, 
             articles.author, 
             articles.created_at, 
             articles.votes, 
             articles.article_img_url
```
Comment (Harrison365) [inline models/select-all-articles.model.js:30]: You should only need to group by one of these.

File: models/select-all-articles.model.js [hunk 4]
```diff
@@ -12,16 +13,26 @@ exports.selectAllArticles = async () => {
...
-        ORDER BY articles.created_at DESC;`
-    );
-    return articles.rows;
+        ORDER BY articles.created_at DESC;`;
+
+    const result = await db.query(queryString, queryArray);
+    if (result.rows.length === 0) {
+        return Promise.reject({statusCode: 404, message: `No articles for topic: ${topic}`});
+    }
```
Comment (Harrison365) [inline models/select-all-articles.model.js:36]: This may cause issues when we add our tests above as an empty array here isn't strictly an error. This is because a topic may exist but have no articles we still want to return an array back.
